qiskit_algorithms/amplitude_estimators/estimation_problem.py

```diff
@@ -190,15 +191,16 @@ def grover_operator(self) -> QuantumCircuit | None:
         oracle.h(self.objective_qubits[-1])
 
         # construct the grover operator
-        return GroverOperator(oracle, self.state_preparation)
+        return grover_operator_builder(oracle, self.state_preparation)
```

Comment:
We could use `qiskit.__version__` to check whether to return `GroverOperator` or `grover_operator` here. This should also be documented then

Reply:
In order not to introduce a breaking change then, is it OK if the function uses `GroverOperator` if the Qiskit version is `2.1.2` or older, and otherwise uses `grover_operator`? That way, it would prevent a breaking change from users upgrading `qiskit-algorithms` without upgrading `qiskit`.

This of course assume that the next `qiskit-algorithms` release is done before Qiskit `2.2.0` is out! Or we could test for `< 3.0`, since this is when `GroverOperator` is removed if I'm not mistaken. What are your views on this?

Reply:
Good questions -- my preference would be to use the `grover_operator` function whenever possible (i.e. `qiskit>=1.3`) since it's more efficient than the `GroverOperator`. But maybe it's safest to just sell this as bugfix and start using `grover_operator` with `qiskit>=2.1` since the `GroverOperator` was deprecated in 2.1.